File: 02_ProgrammingFundamentals/_11_RegularExpressions/_01_Lecture/_03_MatchDates.java
package _02_ProgrammingFundamentals._11_RegularExpressions._01_Lecture;

import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class _03_MatchDates {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input value
        String dates = scan.nextLine();

        // Add variables
        String regex = "\\b(?<day>\\d{2})(?<separator>[\\/\\-.])(?<month>[A-Z][a-z]{2})\\2(?<year>\\d{4})\\b";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(dates);

        // Match dates and print result
        while (matcher.find()) {
            String day = matcher.group("day");
            String month = matcher.group("month");
            String year = matcher.group("year");
            
            System.out.printf("Day: %s, Month: %s, Year: %s%n", day, month, year);
        }
        
        scan.close();
    }
}

File: 24_C#Fundamentals/_09_RegularExpressions/_01_Lecture/_03_MatchDates/_03_MatchDates.cs
using System;
using System.Text.RegularExpressions;

public class _03_MatchDates
{
    public static void Main(string[] args)
    {
        // Read input value
        string dates = Console.ReadLine();

        // Add variables
        string regex = "\\b(?<day>\\d{2})(?<separator>[\\/\\-.])(?<month>[A-Z][a-z]{2})\\2(?<year>\\d{4})\\b";
        Regex pattern = new Regex(regex);
        Match matcher = pattern.Match(dates);

        // Match dates and print result
        while (matcher.Success)
        {
            string day = matcher.Groups["day"].Value;
            string month = matcher.Groups["month"].Value;
            string year = matcher.Groups["year"].Value;

            Console.WriteLine($"Day: {day}, Month: {month}, Year: {year}");
            matcher = matcher.NextMatch();
        }
    }
}
